### ISSUE TITLE
Colon triggers MD051 - request for better error message

### ISSUE THREAD

**Issue by Ben-Voris** [DavidAnson/markdownlint#2080]
```markdown
# test

[link](#1302)

## 13:02
```

Although this is valid GitHub flavored markdown, which is what <https://github.com/DavidAnson/markdownlint/blob/v0.40.0/doc/md051.md> says this rule is meant to handle, this produces the warning `MD051/link-fragments: Link fragments should be valid.`

This was previously reported in <https://github.com/DavidAnson/markdownlint/issues/1959>

So, this is a request to update the text of [MD051](https://github.com/DavidAnson/markdownlint/blob/v0.40.0/doc/md051.md) to state that, although GitHub-flavored markdown accepts this link, the colon will trigger this warning. I.e., one should not need to search issues to discover this apparently intentional behavior. Fwiw, "micromark" is new to me, so I found this behavior surprising. Even if I had heard of "micromark", I wouldn't have know that it is related to markdown lint. 

```text
Markdownlink version 0.61.2
Version: 1.120.0 (Universal)
Commit: 0958016b2af9f09bb4257e0df4a95e2f90590f9f
Date: 2026-05-12T20:17:22Z
Electron: 39.8.8
ElectronBuildId: 13870025
Chromium: 142.0.7444.265
Node.js: 22.22.1
V8: 14.2.231.22-electron.0
OS: Darwin arm64 25.5.0
```

**Comment by DavidAnson** [comment]
Researching this, I decided that inline directive syntax was too permissive. It turns out, I'm not the only one who came to that conclusion. However, the `micromark` folks do not seem to agree. If you want to read an excellent description of the issue, please see: <https://github.com/micromark/micromark-extension-directive/issues/33>. I'm not sure what I want to do about this yet. I don't want to change directive behavior, YET the `:text` scenario seems too common to ignore.

**Comment by Ben-Voris** [comment]
This disconnect is why I suggested updating the descriptive text of the warning on GitHub. For me, although I read a warning that escaping colons in the header text might not be portable (though I can’t remember where), it worked for me. On May 27, 2026, at 23:13, David Anson ***@***.***> wrote:﻿DavidAnson left a comment (DavidAnson/markdownlint#2080)
Researching this, I decided that inline directive syntax was too permissive. It turns out, I'm not the only one who came to that conclusion. However, the micromark folks do not seem to agree. If you want to read an excellent description of the issue, please see: micromark/micromark-extension-directive#33. I'm not sure what I want to do about this yet. I don't want to change directive behavior, YET the :text scenario seems too common to ignore.

**Comment by DavidAnson** [comment]
I'm looking into disabling inline directives instead.
